src/main/java/frc/team2412/robot/subsystem/IndexSubsystem.java

```diff
@@ -1,11 +1,224 @@
+import static frc.team2412.robot.subsystem.IndexSubsystem.IndexConstants.IndexPositionState.*;
 
+import com.ctre.phoenix.motorcontrol.SupplyCurrentLimitConfiguration;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import edu.wpi.first.wpilibj.util.Color;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.team2412.robot.util.MultiplexedColorSensor;
+import io.github.oblarg.oblog.annotations.Log;
+
+import com.ctre.phoenix.motorcontrol.NeutralMode;
 
 public class IndexSubsystem extends SubsystemBase {
-    public IndexSubsystem(WPI_TalonFX indexMotor) {
 
+    // Constants
+
+    public static class IndexConstants {
+
+        public static Alliance teamColor = DriverStation.getAlliance();
+
+        // Color Sensor values
+
+        public static Color BLUE_CARGO_COLOR = new Color(0, 0, 1);
+        public static Color RED_CARGO_COLOR = new Color(1, 0, 0);
+
+        // Index Motor Speeds
+
+        public static final double INDEX_IN_SPEED = 0.5; // will change values later
+        public static final double INDEX_OUT_SPEED = -0.5; // will also change later
+
+        // Proximity Threshold
+        public static final double PROXIMITY_THRESHOLD = 1000; // value not 1500, to be determined actual
+                                                                // value
+
+        // Index Motor States
+
+        public static enum IndexMotorState {
+            IN, OUT, STOPPED;
+        }
+
+        // Position States
+
+        public static enum IndexPositionState {
+            HAS_BALL, HAS_NO_BALL;
+        }
+
+        // The current limit
+        public static final SupplyCurrentLimitConfiguration MAX_MOTOR_CURRENT = new SupplyCurrentLimitConfiguration(
+                true, 40, 40, 500);
+
+    }
+
+    // Define Hardware
+
+    private final MultiplexedColorSensor ingestColorSensor;
+    private final MultiplexedColorSensor feederColorSensor;
+
+    private final WPI_TalonFX ingestMotor;
+    private final WPI_TalonFX feederMotor;
+
+    // States
+
+    private IndexMotorState ingestMotorState;
+    private IndexMotorState feederMotorState;
+
+    private IndexPositionState ingestBallState;
+    private IndexPositionState feederBallState;
+
+    // Constructor
+
+    public IndexSubsystem(WPI_TalonFX firstMotor, WPI_TalonFX secondMotor, MultiplexedColorSensor firstColorSensor,
+            MultiplexedColorSensor secondColorSensor) {
+        this.ingestMotor = firstMotor;
+        this.feederMotor = secondMotor;
+        this.ingestColorSensor = firstColorSensor;
+        this.feederColorSensor = secondColorSensor;
+
+        this.ingestMotor.configFactoryDefault();
+        this.feederMotor.configFactoryDefault();
+
+        this.ingestMotor.setNeutralMode(NeutralMode.Coast);
+        this.feederMotor.setNeutralMode(NeutralMode.Coast);
```

Comment:
Should be brake mode, so the balls stop when the motors stop (coast mode will let the balls be pushed around)